### PR TITLE
feat: implement stubs

### PR DESCRIPTION
**Pull request by hashableric**
### Description
 
The implements of hyperlane agent cosmos-stub

### Drive-by changes

I unify the `{workspace = true}` conventions of `cargo.toml`s

### Related issues

https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2466

### Testing

manual

### CODE REVIEW

**Comment by byeongsu-hong** [comment]
It cannot be determined which job has done in this PR with the title imo

File: rust/chains/hyperlane-cosmos/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+sha256 = { workspace = true }
+hex = { workspace = true }
+
+hyperlane-core = { path = "../../hyperlane-core" }
+
+# These should only be used if it _must_ be used to interop with the inner library,
+# all errors exported from a chain crate should be using thiserror or handrolled to
+# make error handling easier.
+# eyre = "never"
+# anyhow = never
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/Cargo.toml:35]: not sure what's going on here cc @tkporter

File: rust/chains/hyperlane-cosmos/src/mailbox.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+        let response: TxResponse = self
+            .provider
+            .wasm_send(process_message, tx_gas_limit)
+            .await?;
+        Ok(TxOutcome {
+            txid: H256::from_slice(hex::decode(response.txhash).unwrap().as_slice()),
+            executed: response.code == 0,
+            gas_used: U256::from(response.gas_used),
+            gas_price: U256::from(response.gas_wanted),
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/mailbox.rs:197]: gas_wanted and gas_price seem different?

File: rust/chains/hyperlane-cosmos/src/mailbox.rs [hunk 3]
```diff
@@ -0,0 +1,322 @@
...
+            process_message: ProcessMessageRequestInner {
+                message: hex::encode(RawHyperlaneMessage::from(message)),
+                metadata: hex::encode(metadata),
+            },
+        };
+
+        let response: SimulateResponse = self.provider.wasm_simulate(process_message).await?;
+        let result = TxCostEstimate {
+            gas_limit: U256::from(response.gas_info.unwrap().gas_used),
+            gas_price: U256::from(2500),
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/mailbox.rs:217]: where does 2500 come from?

File: rust/chains/hyperlane-cosmos/src/multisig_ism.rs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+                .unwrap()
+                .public_key()
+                .to_bytes(),
+            &prefix,
+        )
+        .unwrap();
+
+        let provider = WasmGrpcProvider::new(
+            address.clone(),
+            private_key,
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/multisig_ism.rs:46]: is there some equivalent of multisig smart contract wallets in cosmwasm?

File: rust/chains/hyperlane-cosmos/src/payloads/general.rs
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/payloads/general.rs:1]: I assume all of these can be generated from cw-hyperlane somehow?

File: rust/chains/hyperlane-cosmos/src/providers/rpc.rs [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+        for (idx, tx) in tx_results.iter().enumerate() {
+            let tx_hash = tx_hash[idx];
+            let mut available = false;
+
+            let mut parse_result: Vec<(T, LogMeta)> = vec![];
+
+            for (log_idx, event) in tx.events.clone().iter().enumerate() {
+                if event.kind.as_str().starts_with(Self::WASM_TYPE)
+                    && event.attributes[0].value == self.address
+                {
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/providers/rpc.rs:105]: any way to filter by address on the query?

File: rust/chains/hyperlane-cosmos/src/routing_ism.rs [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        chain_id: String,
+    ) -> Self {
+        let signer_address = verify::pub_to_addr(
+            SigningKey::from_slice(&private_key)
+                .unwrap()
+                .public_key()
+                .to_bytes(),
+            &prefix,
+        )
+        .unwrap();
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/routing_ism.rs:40]: we don't need a signer here

File: rust/chains/hyperlane-cosmos/src/routing_ism.rs [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+impl RoutingIsm for CosmosRoutingIsm {
+    async fn route(&self, message: &HyperlaneMessage) -> ChainResult<H256> {
+        let payload = IsmRouteRequest {
+            route: IsmRouteRequestInner {
+                message: hex::encode(RawHyperlaneMessage::from(message)),
+            },
+        };
+
+        let data = self.provider.wasm_query(payload, None).await?;
+        let response: IsmRouteRespnose = serde_json::from_slice(&data)?;
```
Comment (yorhodes) [inline rust/chains/hyperlane-cosmos/src/routing_ism.rs:85]: nit:
```suggestion
        let response: IsmRouteResponse = serde_json::from_slice(&data)?;
```

**Review by yorhodes — APPROVED** [review]

**Pull request closed without merging**